splunk-otel-android/src/main/java/com/splunk/rum/RumAttributeAppender.java

```diff
@@ -84,9 +89,18 @@ public void onStart(Context parentContext, ReadWriteSpan span) {
         span.setAttribute(SplunkRum.SCREEN_NAME_KEY, currentScreen);
         CurrentNetwork currentNetwork = connectionUtil.getActiveNetwork();
         span.setAttribute(NET_HOST_CONNECTION_TYPE, currentNetwork.getState().getHumanName());
+        appendNetworkAttributes(span, currentNetwork);
+    }
+
+    @SuppressWarnings("NullAway")
+    static void appendNetworkAttributes(Span span, CurrentNetwork currentNetwork) {
         currentNetwork
                 .getSubType()
                 .ifPresent(subtype -> span.setAttribute(NET_HOST_CONNECTION_SUBTYPE, subtype));
+        span.setAttribute(NET_HOST_CARRIER_NAME, currentNetwork.getCarrierName());
```

Comment:
NullAway complains about these, doesn't it?
I think we should null-check here and follow what the API says, even if we're pretty sure it can handle nulls.

Reply:
Hard meh. Will do.